### PR TITLE
0.2.0+dd.14: rm TCP_NODELAY, query timeout, better retries

### PR DESCRIPTION
**Pull request by dbenamydd**

### CODE REVIEW

File: beaker_extensions/cassandra_cql.py [hunk 1]
```diff
@@ -102,6 +104,7 @@ def __init__(self, namespace, url=None, keyspace=None, column_family=None,
         self.__session = cluster.connect(self.__keyspace_cql_safe)
         self.__ensure_table()
         self.__prepare_statements()
+        self.__session.default_timeout = int(params.get('query_timeout', 10))
```
Comment (sethrosenblum) [inline beaker_extensions/cassandra_cql.py]: What unit is this, and what was it previously?

File: beaker_extensions/nosql.py [hunk 2]
```diff
@@ -54,6 +57,13 @@ def get_creation_lock(self, key):
     def _format_key(self, key):
         return self.namespace + '_'
 
+    def _retry(func):
+        """A little wrapper around the retry lib to pass our settings."""
+        def retry_wrapper(self, *args, **kwargs):
+            return retry_call(func, args, kwargs, tries=self._tries, logger=log)
```
Comment (sethrosenblum) [inline beaker_extensions/nosql.py]: This will log a warning when it has to retry the query?

File: beaker_extensions/cassandra_cql.py (hunk 1 above)
Comment (dbenamydd) [inline beaker_extensions/cassandra_cql.py]: Seconds, I'll clarify.

That's the driver default according to the docs.

File: beaker_extensions/nosql.py (hunk 2 above)
Comment (dbenamydd) [inline beaker_extensions/nosql.py]: Yep.

**Comment by dbenamydd** [comment]
For reference, I tested this by blocking cassandra access locally with `sudo iptables -I INPUT -p tcp --dport 9042 -j DROP; sleep 5; sudo iptables -D INPUT -p tcp --dport 9042 -j DROP`

**Pull request merged**